op-service/dial/active_l2_provider_test.go

```diff
@@ -0,0 +1,683 @@
+
+	// Primary becomes active again, secondary becomes inactive
+	primarySequencer.ExpectSequencerActive(true, nil)
+	secondarySequencer.ExpectSequencerActive(false, nil)
+	secondarySequencer.ExpectClose()
+	ept.ethClients[1].ExpectClose()
+
+	// // Should return to primary
+	thirdSequencerUsed, err := endpointProvider.EthClient(context.Background())
+	require.NoError(t, err)
+	require.Same(t, ept.ethClients[0], thirdSequencerUsed)
+	ept.assertAllExpectations(t)
+}
+
+// TestRollupProvider_InitialActiveSequencerSelection verifies that the ActiveL2RollupProvider
+// selects the active sequencer correctly at the time of creation.
+func TestRollupProvider_InitialActiveSequencerSelection(t *testing.T) {
+	ept := setupEndpointProviderTest(t, 2)
+	primarySequencer := ept.rollupClients[0]
+
+	// Primary active at creation
+	primarySequencer.ExpectSequencerActive(true, nil)
+
+	rollupProvider, err := ept.newActiveL2RollupProvider(0)
+	require.NoError(t, err)
+
+	// Check immediately after creation without additional Active check
+	require.Same(t, primarySequencer, rollupProvider.currentRollupClient)
+	ept.assertAllExpectations(t)
+}
+
+// TestEndpointProvider_InitialActiveSequencerSelection verifies that the ActiveL2EndpointProvider
+// selects the active sequencer correctly at the time of creation.
+func TestEndpointProvider_InitialActiveSequencerSelection(t *testing.T) {
+	ept := setupEndpointProviderTest(t, 2)
+	primarySequencer := ept.rollupClients[0]
+
+	// Primary active at creation
+	primarySequencer.ExpectSequencerActive(true, nil)
+	primarySequencer.ExpectSequencerActive(true, nil) // see comment in other tests about why we expect this twice
+
+	rollupProvider, err := ept.newActiveL2EndpointProvider(0)
+	require.NoError(t, err)
+
+	// Check immediately after creation without additional Active check
+	require.Same(t, primarySequencer, rollupProvider.currentRollupClient)
+	ept.assertAllExpectations(t)
+}
+
+// TestRollupProvider_SelectSecondSequencerIfFirstInactiveAtCreation verifies that if the first sequencer
+// is inactive at the time of ActiveL2RollupProvider creation, the second active sequencer is chosen.
+func TestRollupProvider_SelectSecondSequencerIfFirstInactiveAtCreation(t *testing.T) {
+	ept := setupEndpointProviderTest(t, 2)
+
+	// First sequencer is inactive, second sequencer is active
+	ept.rollupClients[0].ExpectSequencerActive(false, nil)
+	ept.rollupClients[0].ExpectClose()
+	ept.rollupClients[1].ExpectSequencerActive(true, nil)
+
+	rollupProvider, err := ept.newActiveL2RollupProvider(0)
+	require.NoError(t, err)
+
+	require.Same(t, ept.rollupClients[1], rollupProvider.currentRollupClient)
+	ept.assertAllExpectations(t)
+}
+
+// TestRollupProvider_SelectLastSequencerIManyInactiveAtCreation verifies that if all but the last sequencer
+// are inactive at the time of ActiveL2RollupProvider creation, the last active sequencer is chosen.
+func TestRollupProvider_SelectLastSequencerIfManyInactiveAtCreation(t *testing.T) {
```

Comment:
We also want a test where the first 4 sequencers are offline, i.e., not dialable. We might be able to just adjust the test for 2. No need to duplicate test for 2 and 5 seqs.

Reply:
changed `ManyInactive` to `ManyOffline` and changed test flow to make them not dialable rather than errored.